components/mpas-framework/src/tools/registry/gen_inc.c

```diff
@@ -1723,10 +1726,21 @@ int parse_var(FILE *fd, ezxml_t registry, ezxml_t superStruct, ezxml_t currentVa
 		vartimelevs = ezxml_attr(superStruct, "time_levs");
 	}
 
+	time_levs_from_config = false;
 	if(vartimelevs){
-		time_levs = atoi(vartimelevs);
-		if(time_levs < 1){
-			time_levs = 1;
+		if(strncmp(vartimelevs, "namelist:", 9) == 0){
+			time_levs_from_config = true;
+            // set time_levs to 2 to trigger array declarations below
+            time_levs = 2;
```

Comment:
Hi @mark-petersen , what is the reason for not setting here `time_levs` to the number of time levels from the namelist flag?

Reply:
@mark-petersen, indentation here needs to be with tabs, not spaces.

Reply:
@gcapodag, you don't know the number of time levels at compile time when this code is called, so you need a placeholder.

But @mark-petersen, I think the placeholder is confusing and you should rely only on `time_levs_from_config`, not `time_levs > ` for this situation.

Reply:
Thanks. I agree with your recommendations and will not set a dummy `time_levs` for this case. Rather, I can use `time_levs_from_config` as @xylar recommended.